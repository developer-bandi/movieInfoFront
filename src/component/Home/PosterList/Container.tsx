import { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import PosterList from './Presentational';
import imagePreload from '../../../lib/imagePreload';
import { ReducerType } from '../../../store';
import { movePage } from '../../../store/homePoster/Reducer';

const NowShowingContainer = ({ title }: { title: string }) => {
  const dispatch = useDispatch();
  const postersData = useSelector((state: ReducerType) => state.homePoster);
  const [nowPage, setNowPage] = useState(
    title === '현재상영중'
      ? postersData.content.page.nowShowingInfo
      : postersData.content.page.nowCommingInfo
  );

  useEffect(() => {
    if (postersData.content.posterList !== undefined) {
      postersData.content.posterList.nowCommingInfo
        .concat(postersData.content.posterList.nowShowingInfo)
        .forEach((posterData) => {
          imagePreload(posterData.posterPath);
        });
    }
  }, [postersData]);
  //이미지를 미리 로드하여 포스터 전환시 포스터를 빠르게 보여줍니다.

  const changePage = (position: string) => {
    setNowPage(position === 'prev' ? nowPage - 1 : nowPage + 1);
    dispatch(
      movePage({
        kind: title === '현재상영중' ? 'nowShowingInfo' : 'nowCommingInfo',
        page: nowPage + 1,
      })
    );
  };

  return (
    <>
      <PosterList
        postersData={postersData}
        title={title}
        changePage={changePage}
        page={
          postersData.content.page[
            title === '현재상영중' ? 'nowShowingInfo' : 'nowCommingInfo'
          ]
        }
      />
    </>
  );
};

export default NowShowingContainer;
