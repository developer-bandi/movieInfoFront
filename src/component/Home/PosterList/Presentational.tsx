import { FiChevronLeft, FiChevronRight } from 'react-icons/fi';
import nullmovie from '../../../imgs/nullmovie.webp';
import styles from './Style';
import { HomePosterState } from '../../../store/homePoster/Reducer';
import React from 'react';

interface PosterListProps {
  title: string;
  postersData: HomePosterState;
  changePage: (position: string) => void;
  page: number;
}
const PosterListPresentational = ({
  title,
  postersData,
  changePage,
  page,
}: PosterListProps) => {
  if (postersData.loading) {
    return (
      <styles.MainBlock>
        <styles.TitleBlock>
          <styles.Title>{title}</styles.Title>
        </styles.TitleBlock>
        <styles.PosterListBlock>
          <styles.NullButton />
          {new Array(5).fill(0).map((zero, index) => {
            return (
              <styles.NullPosterBlock key={index}></styles.NullPosterBlock>
            );
          })}
          <styles.NullButton />
        </styles.PosterListBlock>
      </styles.MainBlock>
    );
  } else if (postersData.error) {
    return (
      <styles.MainBlock>
        <styles.TitleBlock>
          <styles.Title>{title}</styles.Title>
        </styles.TitleBlock>
        <styles.PosterListBlock>
          <styles.Error>에러 발생</styles.Error>
        </styles.PosterListBlock>
      </styles.MainBlock>
    );
  } else {
    return (
      <styles.MainBlock>
        <styles.TitleBlock>
          <styles.Title>{title}</styles.Title>
        </styles.TitleBlock>
        <styles.PosterListBlock>
          {page === 1 ? (
            <styles.NullButton />
          ) : (
            <styles.PageButton
              onClick={() => changePage('prev')}
              key={`prev${title}`}
            >
              <FiChevronLeft size={40} />
            </styles.PageButton>
          )}
          {postersData.content.posterList !== undefined &&
            postersData.content.posterList[
              title === '현재상영중' ? 'nowShowingInfo' : 'nowCommingInfo'
            ]
              .slice((page - 1) * 5, 5 * page)
              .map((posterData, index) => {
                return (
                  <React.Fragment key={posterData.id}>
                    <styles.PosterBlock to={`/moviedetail/${posterData.id}`}>
                      {posterData.posterPath === null ? (
                        <styles.MovieImg src={nullmovie} alt="notPoster" />
                      ) : (
                        <styles.MovieImg
                          src={`https://image.tmdb.org/t/p/w500${posterData.posterPath}`}
                          alt="poster"
                        ></styles.MovieImg>
                      )}
                      <styles.MovieInfo className="movieinfo">
                        {posterData.overview.length > 170
                          ? posterData.overview.substring(0, 170) + '...'
                          : posterData.overview}
                      </styles.MovieInfo>
                      <styles.MovieRating className="rating">
                        평점 {posterData.voteAverage}
                      </styles.MovieRating>
                      <styles.MoiveName className="moviename">
                        {posterData.title.length > 12
                          ? posterData.title.substring(0, 12) + '...'
                          : posterData.title}
                      </styles.MoiveName>
                    </styles.PosterBlock>
                  </React.Fragment>
                );
              })}
          {
            postersData.content.posterList === undefined ||
            page * 5 >=
              postersData.content.posterList[
                title === '현재상영중' ? 'nowShowingInfo' : 'nowCommingInfo'
              ].length ? (
              <styles.NullButton />
            ) : (
              <styles.PageButton
                onClick={() => changePage('next')}
                key={`next${title}`}
              >
                <FiChevronRight size={40} />
              </styles.PageButton>
            ) /*이후로 넘길수 없으면 빈박스만 렌더링하고 버튼을 누르면 start와 end값에 5를 추가 */
          }
        </styles.PosterListBlock>
      </styles.MainBlock>
    );
  }
};

//이미지를 미리 로드해 두는 함수이다.

export default PosterListPresentational;
