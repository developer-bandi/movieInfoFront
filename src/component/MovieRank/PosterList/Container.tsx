import { useSelector } from 'react-redux';
import PosterListPresentational from './Presentational';
import { ReducerType } from '../../../store';

const PosterListContainer = () => {
  const movieRankList = useSelector((state: ReducerType) => state.movieRank);

  return <PosterListPresentational movieRankList={movieRankList} />;
};

export default PosterListContainer;
