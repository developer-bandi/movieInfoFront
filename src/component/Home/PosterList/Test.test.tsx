import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import PosterList from './Presentational';

describe('PosterList Presentational', () => {
  const PosterInfo = {
    id: 1,
    title: 'test',
    overview: 'testOverview',
    voteAverage: 9.0,
    posterPath: 'testPath',
  };

  it('로딩중', () => {
    const utils = render(
      <PosterList
        title={'현재상영중'}
        postersData={{
          content: {
            page: {
              nowShowingInfo: 1,
              nowCommingInfo: 1,
            },
          },
          loading: true,
          error: false,
        }}
        changePage={jest.fn()}
        page={1}
      />,
      { wrapper: MemoryRouter }
    );
    expect(utils.container).toMatchSnapshot();
    screen.getByText('현재상영중');
  });

  it('에러발생', () => {
    const utils = render(
      <PosterList
        title={'현재상영중'}
        postersData={{
          content: {
            posterList: {
              nowShowingInfo: [PosterInfo],
              nowCommingInfo: [PosterInfo],
              key: '12345',
            },
            page: {
              nowShowingInfo: 1,
              nowCommingInfo: 1,
            },
          },
          loading: false,
          error: true,
        }}
        changePage={jest.fn()}
        page={1}
      />,
      { wrapper: MemoryRouter }
    );
    expect(utils.container).toMatchSnapshot();
    screen.getByText('현재상영중');
    screen.getByText('에러 발생');
  });

  it('정상 작동 첫번째 페이지', () => {
    const utils = render(
      <PosterList
        title={'현재상영중'}
        postersData={{
          content: {
            posterList: {
              nowShowingInfo: new Array(15).fill(0).map((zero, index) => {
                const PosterInfoIdChange = { ...PosterInfo };
                PosterInfoIdChange.id = index + 1;
                return PosterInfoIdChange;
              }),
              nowCommingInfo: new Array(15).fill(0).map((zero, index) => {
                const PosterInfoIdChange = { ...PosterInfo };
                PosterInfoIdChange.id = index + 1;
                return PosterInfoIdChange;
              }),
              key: '12345',
            },
            page: {
              nowShowingInfo: 1,
              nowCommingInfo: 1,
            },
          },
          loading: false,
          error: false,
        }}
        changePage={jest.fn()}
        page={1}
      />,
      { wrapper: MemoryRouter }
    );
    expect(utils.container).toMatchSnapshot();
  });
  it('정상 작동 중간 페이지', () => {
    const utils = render(
      <PosterList
        title={'현재상영중'}
        postersData={{
          content: {
            posterList: {
              nowShowingInfo: new Array(15).fill(0).map((zero, index) => {
                const PosterInfoIdChange = { ...PosterInfo };
                PosterInfoIdChange.id = index + 1;
                return PosterInfoIdChange;
              }),
              nowCommingInfo: new Array(15).fill(0).map((zero, index) => {
                const PosterInfoIdChange = { ...PosterInfo };
                PosterInfoIdChange.id = index + 1;
                return PosterInfoIdChange;
              }),
              key: '12345',
            },
            page: {
              nowShowingInfo: 2,
              nowCommingInfo: 1,
            },
          },
          loading: false,
          error: false,
        }}
        changePage={jest.fn()}
        page={1}
      />,
      { wrapper: MemoryRouter }
    );
    expect(utils.container).toMatchSnapshot();
  });

  it('정상 작동 마지막 페이지', () => {
    const utils = render(
      <PosterList
        title={'현재상영중'}
        postersData={{
          content: {
            posterList: {
              nowShowingInfo: new Array(15).fill(0).map((zero, index) => {
                const PosterInfoIdChange = { ...PosterInfo };
                PosterInfoIdChange.id = index + 1;
                return PosterInfoIdChange;
              }),
              nowCommingInfo: new Array(15).fill(0).map((zero, index) => {
                const PosterInfoIdChange = { ...PosterInfo };
                PosterInfoIdChange.id = index + 1;
                return PosterInfoIdChange;
              }),
              key: '12345',
            },
            page: {
              nowShowingInfo: 3,
              nowCommingInfo: 1,
            },
          },
          loading: false,
          error: false,
        }}
        changePage={jest.fn()}
        page={1}
      />,
      { wrapper: MemoryRouter }
    );
    expect(utils.container).toMatchSnapshot();
  });
});
