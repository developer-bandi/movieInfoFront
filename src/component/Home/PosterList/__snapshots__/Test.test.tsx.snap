// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PosterList/> Presentational 로딩중 1`] = `
<div>
  <section
    class="sc-bdvvtL czXQXW"
  >
    <div
      class="sc-gsDKAQ ivqJhT"
    >
      <h2
        class="sc-dkPtRN dCiWcm"
      >
        현재상영중
      </h2>
    </div>
    <div
      class="sc-hKwDye gUmhIP"
    >
      <div
        class="sc-iqseJM bpmKrU"
      />
      <div
        class="sc-gKclnd eTmFfq"
      />
      <div
        class="sc-gKclnd eTmFfq"
      />
      <div
        class="sc-gKclnd eTmFfq"
      />
      <div
        class="sc-gKclnd eTmFfq"
      />
      <div
        class="sc-gKclnd eTmFfq"
      />
      <div
        class="sc-iqseJM bpmKrU"
      />
    </div>
  </section>
</div>
`;

exports[`<PosterList/> Presentational 에러발생 1`] = `
<div>
  <section
    class="sc-bdvvtL czXQXW"
  >
    <div
      class="sc-gsDKAQ ivqJhT"
    >
      <h2
        class="sc-dkPtRN dCiWcm"
      >
        현재상영중
      </h2>
    </div>
    <div
      class="sc-hKwDye gUmhIP"
    >
      <div
        class="sc-eCImPb dKlIwI"
      >
        에러 발생
      </div>
    </div>
  </section>
</div>
`;

exports[`<PosterList/> Presentational 정상 작동 마지막 페이지 1`] = `
<div>
  <section
    class="sc-bdvvtL czXQXW"
  >
    <div
      class="sc-gsDKAQ ivqJhT"
    >
      <h2
        class="sc-dkPtRN dCiWcm"
      >
        현재상영중
      </h2>
    </div>
    <div
      class="sc-hKwDye gUmhIP"
    >
      <div
        class="sc-iqseJM bpmKrU"
      />
      <a
        class="sc-jRQBWg iYPEWE"
        href="/moviedetail/1"
      >
        <img
          alt="poster"
          class="sc-iCfMLu ljhItf"
          src="https://image.tmdb.org/t/p/w500testPath"
        />
        <div
          class="sc-furwcr jChnMb movieinfo"
        >
          testOverview
        </div>
        <div
          class="sc-pVTFL liMNSY rating"
        >
          평점 
          9
        </div>
        <div
          class="sc-jrQzAO gTgYot moviename"
        >
          test
        </div>
      </a>
      <a
        class="sc-jRQBWg iYPEWE"
        href="/moviedetail/2"
      >
        <img
          alt="poster"
          class="sc-iCfMLu ljhItf"
          src="https://image.tmdb.org/t/p/w500testPath"
        />
        <div
          class="sc-furwcr jChnMb movieinfo"
        >
          testOverview
        </div>
        <div
          class="sc-pVTFL liMNSY rating"
        >
          평점 
          9
        </div>
        <div
          class="sc-jrQzAO gTgYot moviename"
        >
          test
        </div>
      </a>
      <a
        class="sc-jRQBWg iYPEWE"
        href="/moviedetail/3"
      >
        <img
          alt="poster"
          class="sc-iCfMLu ljhItf"
          src="https://image.tmdb.org/t/p/w500testPath"
        />
        <div
          class="sc-furwcr jChnMb movieinfo"
        >
          testOverview
        </div>
        <div
          class="sc-pVTFL liMNSY rating"
        >
          평점 
          9
        </div>
        <div
          class="sc-jrQzAO gTgYot moviename"
        >
          test
        </div>
      </a>
      <a
        class="sc-jRQBWg iYPEWE"
        href="/moviedetail/4"
      >
        <img
          alt="poster"
          class="sc-iCfMLu ljhItf"
          src="https://image.tmdb.org/t/p/w500testPath"
        />
        <div
          class="sc-furwcr jChnMb movieinfo"
        >
          testOverview
        </div>
        <div
          class="sc-pVTFL liMNSY rating"
        >
          평점 
          9
        </div>
        <div
          class="sc-jrQzAO gTgYot moviename"
        >
          test
        </div>
      </a>
      <a
        class="sc-jRQBWg iYPEWE"
        href="/moviedetail/5"
      >
        <img
          alt="poster"
          class="sc-iCfMLu ljhItf"
          src="https://image.tmdb.org/t/p/w500testPath"
        />
        <div
          class="sc-furwcr jChnMb movieinfo"
        >
          testOverview
        </div>
        <div
          class="sc-pVTFL liMNSY rating"
        >
          평점 
          9
        </div>
        <div
          class="sc-jrQzAO gTgYot moviename"
        >
          test
        </div>
      </a>
      <div
        class="sc-kDTinF cIWkgn"
      >
        <svg
          fill="none"
          height="40"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="9 18 15 12 9 6"
          />
        </svg>
      </div>
    </div>
  </section>
</div>
`;

exports[`<PosterList/> Presentational 정상 작동 중간 페이지 1`] = `
<div>
  <section
    class="sc-bdvvtL czXQXW"
  >
    <div
      class="sc-gsDKAQ ivqJhT"
    >
      <h2
        class="sc-dkPtRN dCiWcm"
      >
        현재상영중
      </h2>
    </div>
    <div
      class="sc-hKwDye gUmhIP"
    >
      <div
        class="sc-iqseJM bpmKrU"
      />
      <a
        class="sc-jRQBWg iYPEWE"
        href="/moviedetail/1"
      >
        <img
          alt="poster"
          class="sc-iCfMLu ljhItf"
          src="https://image.tmdb.org/t/p/w500testPath"
        />
        <div
          class="sc-furwcr jChnMb movieinfo"
        >
          testOverview
        </div>
        <div
          class="sc-pVTFL liMNSY rating"
        >
          평점 
          9
        </div>
        <div
          class="sc-jrQzAO gTgYot moviename"
        >
          test
        </div>
      </a>
      <a
        class="sc-jRQBWg iYPEWE"
        href="/moviedetail/2"
      >
        <img
          alt="poster"
          class="sc-iCfMLu ljhItf"
          src="https://image.tmdb.org/t/p/w500testPath"
        />
        <div
          class="sc-furwcr jChnMb movieinfo"
        >
          testOverview
        </div>
        <div
          class="sc-pVTFL liMNSY rating"
        >
          평점 
          9
        </div>
        <div
          class="sc-jrQzAO gTgYot moviename"
        >
          test
        </div>
      </a>
      <a
        class="sc-jRQBWg iYPEWE"
        href="/moviedetail/3"
      >
        <img
          alt="poster"
          class="sc-iCfMLu ljhItf"
          src="https://image.tmdb.org/t/p/w500testPath"
        />
        <div
          class="sc-furwcr jChnMb movieinfo"
        >
          testOverview
        </div>
        <div
          class="sc-pVTFL liMNSY rating"
        >
          평점 
          9
        </div>
        <div
          class="sc-jrQzAO gTgYot moviename"
        >
          test
        </div>
      </a>
      <a
        class="sc-jRQBWg iYPEWE"
        href="/moviedetail/4"
      >
        <img
          alt="poster"
          class="sc-iCfMLu ljhItf"
          src="https://image.tmdb.org/t/p/w500testPath"
        />
        <div
          class="sc-furwcr jChnMb movieinfo"
        >
          testOverview
        </div>
        <div
          class="sc-pVTFL liMNSY rating"
        >
          평점 
          9
        </div>
        <div
          class="sc-jrQzAO gTgYot moviename"
        >
          test
        </div>
      </a>
      <a
        class="sc-jRQBWg iYPEWE"
        href="/moviedetail/5"
      >
        <img
          alt="poster"
          class="sc-iCfMLu ljhItf"
          src="https://image.tmdb.org/t/p/w500testPath"
        />
        <div
          class="sc-furwcr jChnMb movieinfo"
        >
          testOverview
        </div>
        <div
          class="sc-pVTFL liMNSY rating"
        >
          평점 
          9
        </div>
        <div
          class="sc-jrQzAO gTgYot moviename"
        >
          test
        </div>
      </a>
      <div
        class="sc-kDTinF cIWkgn"
      >
        <svg
          fill="none"
          height="40"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="9 18 15 12 9 6"
          />
        </svg>
      </div>
    </div>
  </section>
</div>
`;

exports[`<PosterList/> Presentational 정상 작동 첫번째 페이지 1`] = `
<div>
  <section
    class="sc-bdvvtL czXQXW"
  >
    <div
      class="sc-gsDKAQ ivqJhT"
    >
      <h2
        class="sc-dkPtRN dCiWcm"
      >
        현재상영중
      </h2>
    </div>
    <div
      class="sc-hKwDye gUmhIP"
    >
      <div
        class="sc-iqseJM bpmKrU"
      />
      <a
        class="sc-jRQBWg iYPEWE"
        href="/moviedetail/1"
      >
        <img
          alt="poster"
          class="sc-iCfMLu ljhItf"
          src="https://image.tmdb.org/t/p/w500testPath"
        />
        <div
          class="sc-furwcr jChnMb movieinfo"
        >
          testOverview
        </div>
        <div
          class="sc-pVTFL liMNSY rating"
        >
          평점 
          9
        </div>
        <div
          class="sc-jrQzAO gTgYot moviename"
        >
          test
        </div>
      </a>
      <a
        class="sc-jRQBWg iYPEWE"
        href="/moviedetail/2"
      >
        <img
          alt="poster"
          class="sc-iCfMLu ljhItf"
          src="https://image.tmdb.org/t/p/w500testPath"
        />
        <div
          class="sc-furwcr jChnMb movieinfo"
        >
          testOverview
        </div>
        <div
          class="sc-pVTFL liMNSY rating"
        >
          평점 
          9
        </div>
        <div
          class="sc-jrQzAO gTgYot moviename"
        >
          test
        </div>
      </a>
      <a
        class="sc-jRQBWg iYPEWE"
        href="/moviedetail/3"
      >
        <img
          alt="poster"
          class="sc-iCfMLu ljhItf"
          src="https://image.tmdb.org/t/p/w500testPath"
        />
        <div
          class="sc-furwcr jChnMb movieinfo"
        >
          testOverview
        </div>
        <div
          class="sc-pVTFL liMNSY rating"
        >
          평점 
          9
        </div>
        <div
          class="sc-jrQzAO gTgYot moviename"
        >
          test
        </div>
      </a>
      <a
        class="sc-jRQBWg iYPEWE"
        href="/moviedetail/4"
      >
        <img
          alt="poster"
          class="sc-iCfMLu ljhItf"
          src="https://image.tmdb.org/t/p/w500testPath"
        />
        <div
          class="sc-furwcr jChnMb movieinfo"
        >
          testOverview
        </div>
        <div
          class="sc-pVTFL liMNSY rating"
        >
          평점 
          9
        </div>
        <div
          class="sc-jrQzAO gTgYot moviename"
        >
          test
        </div>
      </a>
      <a
        class="sc-jRQBWg iYPEWE"
        href="/moviedetail/5"
      >
        <img
          alt="poster"
          class="sc-iCfMLu ljhItf"
          src="https://image.tmdb.org/t/p/w500testPath"
        />
        <div
          class="sc-furwcr jChnMb movieinfo"
        >
          testOverview
        </div>
        <div
          class="sc-pVTFL liMNSY rating"
        >
          평점 
          9
        </div>
        <div
          class="sc-jrQzAO gTgYot moviename"
        >
          test
        </div>
      </a>
      <div
        class="sc-kDTinF cIWkgn"
      >
        <svg
          fill="none"
          height="40"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="9 18 15 12 9 6"
          />
        </svg>
      </div>
    </div>
  </section>
</div>
`;

exports[`PosterList Presentational 로딩중 1`] = `
<div>
  <section
    class="sc-bdvvtL czXQXW"
  >
    <div
      class="sc-gsDKAQ ivqJhT"
    >
      <h2
        class="sc-dkPtRN dCiWcm"
      >
        현재상영중
      </h2>
    </div>
    <div
      class="sc-hKwDye gUmhIP"
    >
      <div
        class="sc-iqseJM bpmKrU"
      />
      <div
        class="sc-gKclnd eTmFfq"
      />
      <div
        class="sc-gKclnd eTmFfq"
      />
      <div
        class="sc-gKclnd eTmFfq"
      />
      <div
        class="sc-gKclnd eTmFfq"
      />
      <div
        class="sc-gKclnd eTmFfq"
      />
      <div
        class="sc-iqseJM bpmKrU"
      />
    </div>
  </section>
</div>
`;

exports[`PosterList Presentational 에러발생 1`] = `
<div>
  <section
    class="sc-bdvvtL czXQXW"
  >
    <div
      class="sc-gsDKAQ ivqJhT"
    >
      <h2
        class="sc-dkPtRN dCiWcm"
      >
        현재상영중
      </h2>
    </div>
    <div
      class="sc-hKwDye gUmhIP"
    >
      <div
        class="sc-eCImPb dKlIwI"
      >
        에러 발생
      </div>
    </div>
  </section>
</div>
`;

exports[`PosterList Presentational 정상 작동 마지막 페이지 1`] = `
<div>
  <section
    class="sc-bdvvtL czXQXW"
  >
    <div
      class="sc-gsDKAQ ivqJhT"
    >
      <h2
        class="sc-dkPtRN dCiWcm"
      >
        현재상영중
      </h2>
    </div>
    <div
      class="sc-hKwDye gUmhIP"
    >
      <div
        class="sc-iqseJM bpmKrU"
      />
      <a
        class="sc-jRQBWg iYPEWE"
        href="/moviedetail/1"
      >
        <img
          alt="poster"
          class="sc-iCfMLu ljhItf"
          src="https://image.tmdb.org/t/p/w500testPath"
        />
        <div
          class="sc-furwcr jChnMb movieinfo"
        >
          testOverview
        </div>
        <div
          class="sc-pVTFL liMNSY rating"
        >
          평점 
          9
        </div>
        <div
          class="sc-jrQzAO gTgYot moviename"
        >
          test
        </div>
      </a>
      <a
        class="sc-jRQBWg iYPEWE"
        href="/moviedetail/2"
      >
        <img
          alt="poster"
          class="sc-iCfMLu ljhItf"
          src="https://image.tmdb.org/t/p/w500testPath"
        />
        <div
          class="sc-furwcr jChnMb movieinfo"
        >
          testOverview
        </div>
        <div
          class="sc-pVTFL liMNSY rating"
        >
          평점 
          9
        </div>
        <div
          class="sc-jrQzAO gTgYot moviename"
        >
          test
        </div>
      </a>
      <a
        class="sc-jRQBWg iYPEWE"
        href="/moviedetail/3"
      >
        <img
          alt="poster"
          class="sc-iCfMLu ljhItf"
          src="https://image.tmdb.org/t/p/w500testPath"
        />
        <div
          class="sc-furwcr jChnMb movieinfo"
        >
          testOverview
        </div>
        <div
          class="sc-pVTFL liMNSY rating"
        >
          평점 
          9
        </div>
        <div
          class="sc-jrQzAO gTgYot moviename"
        >
          test
        </div>
      </a>
      <a
        class="sc-jRQBWg iYPEWE"
        href="/moviedetail/4"
      >
        <img
          alt="poster"
          class="sc-iCfMLu ljhItf"
          src="https://image.tmdb.org/t/p/w500testPath"
        />
        <div
          class="sc-furwcr jChnMb movieinfo"
        >
          testOverview
        </div>
        <div
          class="sc-pVTFL liMNSY rating"
        >
          평점 
          9
        </div>
        <div
          class="sc-jrQzAO gTgYot moviename"
        >
          test
        </div>
      </a>
      <a
        class="sc-jRQBWg iYPEWE"
        href="/moviedetail/5"
      >
        <img
          alt="poster"
          class="sc-iCfMLu ljhItf"
          src="https://image.tmdb.org/t/p/w500testPath"
        />
        <div
          class="sc-furwcr jChnMb movieinfo"
        >
          testOverview
        </div>
        <div
          class="sc-pVTFL liMNSY rating"
        >
          평점 
          9
        </div>
        <div
          class="sc-jrQzAO gTgYot moviename"
        >
          test
        </div>
      </a>
      <div
        class="sc-kDTinF cIWkgn"
      >
        <svg
          fill="none"
          height="40"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="9 18 15 12 9 6"
          />
        </svg>
      </div>
    </div>
  </section>
</div>
`;

exports[`PosterList Presentational 정상 작동 중간 페이지 1`] = `
<div>
  <section
    class="sc-bdvvtL czXQXW"
  >
    <div
      class="sc-gsDKAQ ivqJhT"
    >
      <h2
        class="sc-dkPtRN dCiWcm"
      >
        현재상영중
      </h2>
    </div>
    <div
      class="sc-hKwDye gUmhIP"
    >
      <div
        class="sc-iqseJM bpmKrU"
      />
      <a
        class="sc-jRQBWg iYPEWE"
        href="/moviedetail/1"
      >
        <img
          alt="poster"
          class="sc-iCfMLu ljhItf"
          src="https://image.tmdb.org/t/p/w500testPath"
        />
        <div
          class="sc-furwcr jChnMb movieinfo"
        >
          testOverview
        </div>
        <div
          class="sc-pVTFL liMNSY rating"
        >
          평점 
          9
        </div>
        <div
          class="sc-jrQzAO gTgYot moviename"
        >
          test
        </div>
      </a>
      <a
        class="sc-jRQBWg iYPEWE"
        href="/moviedetail/2"
      >
        <img
          alt="poster"
          class="sc-iCfMLu ljhItf"
          src="https://image.tmdb.org/t/p/w500testPath"
        />
        <div
          class="sc-furwcr jChnMb movieinfo"
        >
          testOverview
        </div>
        <div
          class="sc-pVTFL liMNSY rating"
        >
          평점 
          9
        </div>
        <div
          class="sc-jrQzAO gTgYot moviename"
        >
          test
        </div>
      </a>
      <a
        class="sc-jRQBWg iYPEWE"
        href="/moviedetail/3"
      >
        <img
          alt="poster"
          class="sc-iCfMLu ljhItf"
          src="https://image.tmdb.org/t/p/w500testPath"
        />
        <div
          class="sc-furwcr jChnMb movieinfo"
        >
          testOverview
        </div>
        <div
          class="sc-pVTFL liMNSY rating"
        >
          평점 
          9
        </div>
        <div
          class="sc-jrQzAO gTgYot moviename"
        >
          test
        </div>
      </a>
      <a
        class="sc-jRQBWg iYPEWE"
        href="/moviedetail/4"
      >
        <img
          alt="poster"
          class="sc-iCfMLu ljhItf"
          src="https://image.tmdb.org/t/p/w500testPath"
        />
        <div
          class="sc-furwcr jChnMb movieinfo"
        >
          testOverview
        </div>
        <div
          class="sc-pVTFL liMNSY rating"
        >
          평점 
          9
        </div>
        <div
          class="sc-jrQzAO gTgYot moviename"
        >
          test
        </div>
      </a>
      <a
        class="sc-jRQBWg iYPEWE"
        href="/moviedetail/5"
      >
        <img
          alt="poster"
          class="sc-iCfMLu ljhItf"
          src="https://image.tmdb.org/t/p/w500testPath"
        />
        <div
          class="sc-furwcr jChnMb movieinfo"
        >
          testOverview
        </div>
        <div
          class="sc-pVTFL liMNSY rating"
        >
          평점 
          9
        </div>
        <div
          class="sc-jrQzAO gTgYot moviename"
        >
          test
        </div>
      </a>
      <div
        class="sc-kDTinF cIWkgn"
      >
        <svg
          fill="none"
          height="40"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="9 18 15 12 9 6"
          />
        </svg>
      </div>
    </div>
  </section>
</div>
`;

exports[`PosterList Presentational 정상 작동 첫번째 페이지 1`] = `
<div>
  <section
    class="sc-bdvvtL czXQXW"
  >
    <div
      class="sc-gsDKAQ ivqJhT"
    >
      <h2
        class="sc-dkPtRN dCiWcm"
      >
        현재상영중
      </h2>
    </div>
    <div
      class="sc-hKwDye gUmhIP"
    >
      <div
        class="sc-iqseJM bpmKrU"
      />
      <a
        class="sc-jRQBWg iYPEWE"
        href="/moviedetail/1"
      >
        <img
          alt="poster"
          class="sc-iCfMLu ljhItf"
          src="https://image.tmdb.org/t/p/w500testPath"
        />
        <div
          class="sc-furwcr jChnMb movieinfo"
        >
          testOverview
        </div>
        <div
          class="sc-pVTFL liMNSY rating"
        >
          평점 
          9
        </div>
        <div
          class="sc-jrQzAO gTgYot moviename"
        >
          test
        </div>
      </a>
      <a
        class="sc-jRQBWg iYPEWE"
        href="/moviedetail/2"
      >
        <img
          alt="poster"
          class="sc-iCfMLu ljhItf"
          src="https://image.tmdb.org/t/p/w500testPath"
        />
        <div
          class="sc-furwcr jChnMb movieinfo"
        >
          testOverview
        </div>
        <div
          class="sc-pVTFL liMNSY rating"
        >
          평점 
          9
        </div>
        <div
          class="sc-jrQzAO gTgYot moviename"
        >
          test
        </div>
      </a>
      <a
        class="sc-jRQBWg iYPEWE"
        href="/moviedetail/3"
      >
        <img
          alt="poster"
          class="sc-iCfMLu ljhItf"
          src="https://image.tmdb.org/t/p/w500testPath"
        />
        <div
          class="sc-furwcr jChnMb movieinfo"
        >
          testOverview
        </div>
        <div
          class="sc-pVTFL liMNSY rating"
        >
          평점 
          9
        </div>
        <div
          class="sc-jrQzAO gTgYot moviename"
        >
          test
        </div>
      </a>
      <a
        class="sc-jRQBWg iYPEWE"
        href="/moviedetail/4"
      >
        <img
          alt="poster"
          class="sc-iCfMLu ljhItf"
          src="https://image.tmdb.org/t/p/w500testPath"
        />
        <div
          class="sc-furwcr jChnMb movieinfo"
        >
          testOverview
        </div>
        <div
          class="sc-pVTFL liMNSY rating"
        >
          평점 
          9
        </div>
        <div
          class="sc-jrQzAO gTgYot moviename"
        >
          test
        </div>
      </a>
      <a
        class="sc-jRQBWg iYPEWE"
        href="/moviedetail/5"
      >
        <img
          alt="poster"
          class="sc-iCfMLu ljhItf"
          src="https://image.tmdb.org/t/p/w500testPath"
        />
        <div
          class="sc-furwcr jChnMb movieinfo"
        >
          testOverview
        </div>
        <div
          class="sc-pVTFL liMNSY rating"
        >
          평점 
          9
        </div>
        <div
          class="sc-jrQzAO gTgYot moviename"
        >
          test
        </div>
      </a>
      <div
        class="sc-kDTinF cIWkgn"
      >
        <svg
          fill="none"
          height="40"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="9 18 15 12 9 6"
          />
        </svg>
      </div>
    </div>
  </section>
</div>
`;
